### ISSUE TITLE
More on handling Postgres environmental variables in settings.py

### ISSUE THREAD

**Issue by enoren5** [enoren5/tarot_juicer#84]
### Issue 1.
Here is the `DATABASES` environment variable inside my `settings.py` (lines  92-108):

```
SELECTED_DB = ""

VALUE = os.getenv('SELECT_DB')

if VALUE == "0":
   SELECTED_DB = "HEROKU_POSTGRESQL_SILVER_URL"
elif VALUE == "1":
   SELECTED_DB = "HEROKU_POSTGRESQL_NAVY_URL"

DATABASES = {
   'default': dj_database_url.config(
       env=SELECTED_DB,
       default='sqlite:///'+os.path.join(BASE_DIR, 'db.sqlite3'),
       conn_max_age=600)
   }

print(DATABASES)
```
'SELECT_DB' is an environmental variable (dictionary) declared inside Heroku’s dashboard which is paired with the value specified as an integer, either 1 or 0. On the next line, if the `VALUE` is 0, then dj_database_url.configures the db to be `HEROKU_POSTGRESQL_SILVER_URL`. If the value is 1, then dj_database_url.configures the db to be, `HEROKU_POSTGRESQL_NAVY_URL`. These are references to Postgres URIs which are also formed as configuration variables showing inside the Heroku dashboard. The first “SILVER” db contains Lorem Ipsum placeholder content and the second “NAVY” db contains meaningful blog content posted by the end user. This is my general understanding of what is going on here.

The problem with all of the above is that [Heroku Postgres supports flexible and dynamic on-the-fly provisioning for creating, spinning up, spinning down, swapping around different apps, backing up, and restoring db silos](https://devcenter.heroku.com/articles/heroku-postgresql#provisioning-heroku-postgres).  As you can read at that heading in the Heroku guide it says that to initialize (create) a fresh db, in the Heroku shell, you run: `(local venv) $ heroku addons:create heroku-postgresql:hobby-dev` 

To view my current Postgres db addon composition and structure, you can use this command:
```
(local venv)  $ heroku pg:info
 ›   Warning: heroku update available from 7.47.11 to 7.47.12.
=== HEROKU_POSTGRESQL_NAVY_URL, DATABASE_URL
Plan:                  Hobby-dev
Status:                Available
Connections:           0/20
PG Version:            12.5
Created:               2021-01-09 20:43 UTC
Data Size:             10.0 MB
Tables:                21
Rows:                  110/10000 (In compliance)
Fork/Follow:           Unsupported
Rollback:              Unsupported
Continuous Protection: Off
Add-on:                postgresql-real-1

=== HEROKU_POSTGRESQL_ROSE_URL
Plan:                  Hobby-dev
Status:                Available
Connections:           0/20
PG Version:            13.2
Created:               2021-02-26 17:21 UTC
Data Size:             7.9 MB
Tables:                0
Rows:                  0/10000 (In compliance) - refreshing
Fork/Follow:           Unsupported
Rollback:              Unsupported
Continuous Protection: Off
Add-on:                postgresql-adjacent-58092

=== HEROKU_POSTGRESQL_SILVER_URL
Plan:                  Hobby-dev
Status:                Available
Connections:           0/20
PG Version:            12.6
Created:               2020-11-19 10:32 UTC
Data Size:             9.6 MB
Tables:                21
Rows:                  84/10000 (In compliance)
Fork/Follow:           Unsupported
Rollback:              Unsupported
Continuous Protection: Off
Add-on:                postgresql-lorem-1
```

To designate a primary database, you can use this command:

`(local venv) $ heroku pg:promote HEROKU_POSTGRESQL_SILVER_URL` (or whichever other db name is available as it appears in the list above). Here in the command line is where  I should swap out / switch between databases and not statically coded inside settings.py (like it currently is in my first snippet above).

With the newly created ROSE db instance, the `settings.py` is out of date. I would need to add another `elif` condition which specifies a `SELECTED_DB` (as described above) as "HEROKU_POSTGRESQL_ROSE_URL" with a value of 2, and then add that config variables inside the Heroku dashboard manually. That’s not ideal, right? The db environmental variable should allow the Heroku sysadmin to change between db’s using the `(local venv) $ heroku pg:promote <db name>`

### Issue 2:
I should be able to import (access) remote aws Postgres db’s locally by specifying the remote URL in `settings.py` in this format: `postgres://USER:PASSWORD@HOST:PORT/NAME`. [This feature is described in John Elder’s Udemy course specified here](https://youtu.be/UnTCkHiJcO0 ).

This feature was actually already implemented in PR #79 but was lost and removed in subsequent PR’s (#80. #81, #82). This needs to be re-implemented. I tried implementing it myself just now by removing the VALUE/SELECTED switches and replacing them with this:

```
DATABASE_URL = "" # since this is the default handle as you can see above in the output for the command: $ heroku pg:info

DATABASES = {
   'default': dj_database_url.config(
       env=DATABASE_URL,
       default='sqlite:///'+os.path.join(BASE_DIR, 'db.sqlite3'),
       conn_max_age=600)
   }
```
But this throws an OperationalError. 

### To summarize, here is the takeaway: 
The DATABASES environmental variable needs to be reconfigured inside `settings.py` to essentially enable:
**1.** the Heroku sysadmin to swap db’s using the `$ heroku pg:promote` command rather than updating settings.py every time a new db is spinned up/down.
**2.** the web developer to reference the remote db locally

**Comment by enoren5** [comment]
Hi @UmarGit! The OperationalError at /portal is resolved in PR #85 + #86. Django is deployed and this traceback is gone. Thank you for this. However the other component to my issue is how to refer to (by exporting) the aws ec2 instance locally. For example, when running: 

> `(local venv) $ export DATABASE_URL='postgres://jckmeibmxypuvn:11033b956dd9cc6011e8efb7d23d56bb277bef7e9634a31ae0f0b304f34e9667@ec2-3-231-241-17.compute-1.amazonaws.com:5432/d9eds0cfpna1li'`

...my local Django environment is still serving db.sqlite3.  I even tried exiting my local dev server, invoking  `$ unset DATABASE_URL` and then re-invoking the `export` command (above) and re-running the server.  My local db remains sqlite. 

How do I properly export this DATABASE_URL locally?

**Comment by UmarGit** [comment]
> DATABASE_URL

Brother are you using windows ?

**Comment by enoren5** [comment]
> > DATABASE_URL
> 
> Brother are you using windows ?

I run Manjaro Linux locally

**Comment by UmarGit** [comment]
Hey @enoren5 , brother you can run this command to check if `DATABASE_URL`, properly setup.
 
```bash
echo $DATABASE_URL
```

**Comment by enoren5** [comment]
Here it is:
> (local venv) $ echo $DATABASE_URL
> postgres://jckmeibmxypuvn:11033b956dd9cc6011e8efb7d23d56bb277bef7e9634a31ae0f0b304f34e9667@ec2-3-231-241-17.compute-1.amazonaws.com:5432/d9eds0cfpna1li

**Comment by enoren5** [comment]
I even tried with other db instances like this:

> (local venv) $ export DATABASE_URL='postgres://xkdbsiijoknylq:c9f9d6cdb11bb69b34ba65ce4bb845cf1d14f905f1aba539d5ecd6515f1da19d@ec2-54-84-98-18.compute-1.amazonaws.com:5432/dcjnf0h34cfsc5'
> (local venv) $ echo $DATABASE_URL
> postgres://xkdbsiijoknylq:c9f9d6cdb11bb69b34ba65ce4bb845cf1d14f905f1aba539d5ecd6515f1da19d@ec2-54-84-98-18.compute-1.amazonaws.com:5432/dcjnf0h34cfsc5
> 

...yet the local db remains db.sqlite3, even after refreshing and trying to navigate other pages

**Comment by enoren5** [comment]
I can confirm that this issue has been resolved in PR's #85 and #86. Based on @UmarGit's comments, I have consolidated them and expanded upon them in the README.md at [this bookmark here](https://github.com/enoren5/tarot_juicer#handling-db-remote-instances-but-locally). The README.md further explains how to switch remote postgres data from the cloud to appear locally. 

Thank you Umar for your hard work.
